src/diffusers/pipelines/marigold/pipeline_marigold_depth.py

```diff
@@ -0,0 +1,1246 @@
+                    k.dim() == 4 and k.shape[0] == ensemble_size and k.shape[1] == self.latent_space_size
+                    for k in input_latent
+                ):
+                    raise ValueError(
+                        f"Input latent must be 4-dimensional with shape [E,{self.latent_space_size},h,w], "
+                        f"where E is the requested ensemble_size."
+                    )
+                if len(input_latent) != num_images:
+                    raise ValueError(
+                        f"The numbers of input images ({num_images}) and latents ({len(input_latent)}) "
+                        f"are not compatible."
+                    )
+            else:
+                raise ValueError(f"Unsupported latent type: {type(input_latent)}.")
+        if input_image_stacked ^ input_latent_stacked:
+            logger.warning("Different stacking of input images and latents might be a sign of undesired behavior.")
+
+        if generator is not None:
+            if isinstance(generator, torch.Generator):
+                generator = [generator]
+            if isinstance(generator, list) and not isinstance(generator, tuple):
+                if len(generator) == 1 and num_images > 1:
+                    generator = generator * num_images
+                if len(generator) != num_images:
+                    raise ValueError(
+                        f"The numbers of input images ({num_images}) and generators ({len(generator)}) "
+                        f"are not compatible."
+                    )
+            else:
+                raise ValueError(f"Unsupported generator type: {type(generator)}.")
+
+        # Prepare the empty text embedding. In the future, remove text modules completely
+        if self.empty_text_embedding is None:
+            self.encode_empty_text()
+
+        out = []
+        with self.progress_bar(total=num_images * ensemble_size * denoising_steps) as progress_bar:
+            for i in range(num_images):
+                out.append(
+                    self.process_image(
+                        input_image[i],
+                        input_latent[i] if input_latent is not None else None,
+                        generator[i] if generator is not None else None,
+                        denoising_steps,
+                        ensemble_size,
+                        processing_resolution,
+                        match_input_resolution,
+                        resample_method_input,
+                        resample_method_output,
+                        batch_size,
+                        check_input,
+                        ensembling_kwargs,
+                        output_prediction_format,
+                        output_visualization,
+                        output_visualization_kwargs,
+                        output_uncertainty,
+                        output_latent,
+                        progress_bar,
+                    )
+                )
```

Comment:
Can we make this block of code slightly more lucid to follow? More akin to how it's usually done in other official pipelines.

Reply:
Could you please point to an example? Other pipelines I saw that take List[torch.Tensor] as input conditions usually rely on these images being of compatible dimensions to stack them. This is not the case supported in this code: the List we accept can contain images of varying resolutions. I'm not opposed to changing the permitted input type, either.

Reply:
Fixed